js/VideoPlayer_html5Captionator.js

```diff
@@ -53,25 +53,28 @@ https://source.fluidproject.org/svn/LICENSE.txt
             "text/vtt": "fluid.videoPlayer.html5Captionator.createVttTrack"
         }
     });
-    
-    
+
+
     var bindCaptionatorModel = function (that) {
         var elPaths = that.options.elPaths;
         that.applier.modelChanged.addListener(elPaths.currentCaptions, that.refreshCaptions);
         that.applier.modelChanged.addListener(elPaths.displayCaptions, that.refreshCaptions);
     };
-    
+
     fluid.videoPlayer.html5Captionator.hideAllTracks = function (tracks) {
         fluid.each(tracks, function (trackEl) {
             trackEl.track.mode = "disabled";
         });
     };
-    
+
     fluid.videoPlayer.html5Captionator.showCurrentTrack = function (currentCaptions, tracks, captionSources) {
-        fluid.each(captionSources, function (element, key) {
-            var currentState = $.inArray(key, currentCaptions) === -1 ? "disabled" : "showing";
-            var track = tracks[key].track;
-            track.mode =  track[currentState.toUpperCase()] || currentState;
+        fluid.videoPlayer.html5Captionator.hideAllTracks(tracks);
+        fluid.each(currentCaptions, function (capIndex, index) {
+            var id = captionSources[capIndex].id;
+            var track = fluid.find(tracks, function (track, index) {
+                return (track.id == id) ? track : undefined;
+            });
+            track.track.mode =  "showing";
```

Comment:
@jobara, thanks for the pointer to fluid.byId(). Regarding the array: IIRC, yes, it's an array because we imagine that ultimately, we might support multiple captions or transcripts simultaneously. Perhaps when we update the VP to the latest Infusion (which will require a significant re-write, I think), we can reconsider this. It would likely simplify a lot to NOT use an array.